include/uam/workers/scip_worker.h

```diff
@@ -0,0 +1,222 @@
+    const std::string& input,
+    const std::string& first = ":",
+    const std::string& last = ";")
+  {
+    std::string result = "";
+    size_t first_pos = input.find(first);
+
+    first_pos = first_pos == std::string::npos ? first_pos : first_pos + 1;
+    if (first_pos != std::string::npos)
+    {
+      auto last_pos = input.find(last, first_pos);
+      if (last_pos < first_pos)
+      {
+        return result;
+      }
+      result = input.substr(first_pos, last_pos - first_pos);
+    }
+    return result;
+  }
+
+private:
+  /**
+   * @brief The packet callback
+   */
+  PacketEventCallback callback_;
+};
+
+/**
+ * @brief PP command worker
+ */
+class PPWorker : public SCIPWorker<PPWorker, scip_protocol::PPReplyLineIndex::NR_LINES, scip_protocol::PPReply>
+{
+public:
+  /**
+   * @brief Default C'tor
+   */
+  PPWorker() : SCIPWorker<PPWorker, scip_protocol::PPReplyLineIndex::NR_LINES, scip_protocol::PPReply>() {}
+
+  /**
+   * @brief Decode raw reply into Reply
+   * @param[in] raw_reply - Raw reply (array of strings)
+   * @return[out] Decoded reply if success, std::nullopt otherwise
+   */
+  std::optional<Reply> decode(const RawReply& raw_reply) const
+  {
+    if (raw_reply.size() != scip_protocol::PPReplyLineIndex::NR_LINES)
+    {
+      ROS_ERROR_STREAM(
+        "Invalid PP response with only: " << raw_reply.size()
+                                          << " line. Expected: " << scip_protocol::PPReplyLineIndex::NR_LINES);
+      return std::nullopt;
+    }
+
+    Reply reply;
+    bool failed = !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::MIN_DISTANCE), reply.min_distance);
+    failed = failed || !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::MAX_DISTANCE), reply.max_distance);
+    failed = failed ||
+             !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::ANGULAR_RESOLUTION), reply.angular_resolution);
+    failed = failed || !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::STARTING_STEP), reply.first_step);
+    failed = failed || !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::END_STEP), reply.last_step);
+    failed = failed ||
+             !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::STEP_FRONT_DIRECTION), reply.front_data_step);
+    failed = failed || !decodeField(raw_reply.at(scip_protocol::PPReplyLineIndex::RPM), reply.rpm);
+    if (failed)
+    {
+      ROS_ERROR_STREAM("Failed To decode PP reply params");
+      return std::nullopt;
+    }
+    return reply;
+  }
+
+  /**
+   * @brief Retrieve the command
+   *
+   * @return The scip PP command
+   */
+  inline const std::string getCommand() const { return "PP\n"; }
+
+private:
+  /**
+   * @brief Transform string into an integer value
+   *
+   * @param[in] field - Input string
+   * @param[out] value - Output value
+   * @return true if transformation was successful, false otherwise
+   */
+  bool decodeField(const std::string& field, int& value) const
```

Comment:
I think it will be similar but I haven't checked for other scip commands. This is only used in the beginning to get the scan params (I should validate crc but I am not doing that for this command).